### ISSUE TITLE
pool_stride Bug ! ! ! !

### ISSUE THREAD

**Issue by Syndrome777** [dnouri/nolearn#114]
Hello, everybody,
My OS is Windows 8.1.
And I find there is a bug:
The net6 is my model.
when I don't use pool1_stride or pool1_stride == pool1_pool_size, everything is okay, the coder runs on gpu, it will take 20s for a iteration.
But when I use pool1_stride = (1, 1) or pool1_stride != pool1_pool_size, it will take 1800s for a iteration.

So maybe something is wrong

	net6 = NeuralNet(
		layers=[
			('input', layers.InputLayer),
			('conv1', layers.Conv2DLayer),
			('pool1', layers.MaxPool2DLayer),
			('dropout1', layers.DropoutLayer),  # !
			('conv2', layers.Conv2DLayer),
			('pool2', layers.MaxPool2DLayer),
			('dropout2', layers.DropoutLayer),  # !
			('conv3', layers.Conv2DLayer),
			('pool3', layers.MaxPool2DLayer),
			('dropout3', layers.DropoutLayer),  # !
			('conv4', layers.Conv2DLayer),
			('dropout4', layers.DropoutLayer),  # !
			('hidden5', layers.DenseLayer),
			('dropout5', layers.DropoutLayer),  # !
			('hidden6', layers.DenseLayer),
			('output', layers.DenseLayer),
			],

		input_shape=(None, 1, 48, 48),
		conv1_num_filters=32, conv1_filter_size=(3, 3), pool1_pool_size=(3, 3),  pool1_stride = (1, 1),
		dropout1_p=0.1,  # !
		conv2_num_filters=64, conv2_filter_size=(2, 2), pool2_pool_size=(2, 2),
		dropout2_p=0.2,  # !
		conv3_num_filters=64, conv3_filter_size=(2, 2), pool3_pool_size=(2, 2),
		dropout3_p=0.3,  # !
		conv4_num_filters=128, conv4_filter_size=(2, 2),
		dropout4_p=0.4,  # !
		hidden5_num_units=400,
		dropout5_p=0.5,  # !
		hidden6_num_units=250,
		output_num_units=7, output_nonlinearity=softmax,

		# optimization method:
		update=nesterov_momentum,
		# update_learning_rate=0.01,
		# update_momentum=0.9,

		update_learning_rate=0.01,
		update_momentum=0.9,

		regression=False,
		objective_loss_function=categorical_crossentropy,
		batch_iterator_train=BatchIterator(batch_size=64),
		max_epochs=5,
		verbose=1,
		eval_size=0.025,
		)

**Comment by BenjaminBossan** [comment]
For me, this sounds like it could be a Lasagne-problem, since nolearn does not do anything specific with strides. Maybe you can head over to the [Lasagne issue tracker](https://github.com/Lasagne/Lasagne/issues) and report there.

**Comment by dnouri** [comment]
I think Theano's `DownsampleFactorMax` might only have CPU code for max pooling with a custom stride.  I'd try one of the other max pooling layer implementations (dnn, cuda_convnet), and see if you're more lucky regarding speed.
